### PR TITLE
gp/INFRA-671: support Datadog resources

### PR DESCRIPTION
**Pull request by gpuliyar**
Support Datadog resources in the reusable workflow.

### CODE REVIEW

File: .github/workflows/terraform-workflow.yml [hunk 1]
```diff
@@ -142,18 +150,18 @@ jobs:
         continue-on-error: true
         working-directory: ${{ inputs.terraform-working-directory }}
 
-      - uses: envoy/terraform-plan-summary-action@v2
-        if: ${{ steps.terraform-plan.outcome != 'failure' }}
+      - if: ${{ steps.terraform-plan.outcome != 'failure' }}
+        uses: envoy/terraform-plan-summary-action@v2
         with:
-          terrform-plan-file: ${{ inputs.terraform-working-directory }}/tfplan-show.json
+          terraform-plan-file: ${{ inputs.terraform-working-directory }}/tfplan-show.json
```
Comment (gpuliyar) [inline .github/workflows/terraform-workflow.yml:156]: typo fix

File: .github/workflows/terraform-workflow.yml [hunk 2]
```diff
@@ -142,18 +150,18 @@ jobs:
         continue-on-error: true
         working-directory: ${{ inputs.terraform-working-directory }}
 
-      - uses: envoy/terraform-plan-summary-action@v2
-        if: ${{ steps.terraform-plan.outcome != 'failure' }}
+      - if: ${{ steps.terraform-plan.outcome != 'failure' }}
```
Comment (gpuliyar) [inline .github/workflows/terraform-workflow.yml:153]: changed the order. no change to functionality

File: .github/workflows/terraform-workflow.yml [hunk 3]
```diff
@@ -142,18 +150,18 @@ jobs:
...
+          terraform-plan-file: ${{ inputs.terraform-working-directory }}/tfplan-show.json
         continue-on-error: true
 
       - name: Terraform Apply
         id: terraform-apply
         if: ${{ steps.terraform-plan.outcome != 'failure' && inputs.terraform-apply }}
         run: terraform apply -auto-approve tfplan
-        continue-on-error: true
         working-directory: ${{ inputs.terraform-working-directory }}
+        continue-on-error: true
```
Comment (gpuliyar) [inline .github/workflows/terraform-workflow.yml:164]: changed the order. no change to functionality.

File: .github/workflows/terraform-workflow.yml [hunk 4]
```diff
@@ -86,6 +86,12 @@ on:
       cloudflare-account-id:
         description: Cloudflare Account ID to authenticate Cloudflare API.
         required: false
+      datadog-api-key:
```
Comment (gpuliyar) [inline .github/workflows/terraform-workflow.yml:89]: Support Datadog API keys for the `terraform` `plan` and `apply` to work

**Review by antigenius0910 — APPROVED** [review]
LGTM

**Pull request merged**